tests/datatypes/test_str.py

```diff
@@ -276,6 +276,39 @@ def test_len(self):
             print(len(s))
         """)
 
+    def test_center(self):
+        # test ok
+        self.assertCodeExecution("""
+            s = "abc"
+            print(s.center(10, "-"))
+            """)
+        # less length
+        self.assertCodeExecution("""
+            s = "abc"
+            print(s.center(2,"-"))
+            """)
+        # more than one fillChar
+        self.assertCodeExecution("""
+            s = "abc"
+            try:
+                print(s.center(10,"-*"))
+            except TypeError as err:
+                print(err)
+            """)
+        # called method is not string
+        self.assertCodeExecution("""
+            s = 123
+            try:
+                print(s.center(10,"-"))
+            except AttributeError as err:
+                print(err)
+            """)
+        # bool as width
+        self.assertCodeExecution("""
+            s = "abc"
+            print(s.center(True,"-"))
+            """)
```

Comment:
This is a really good set of tests; I have a couple of suggestions:

1. There's a small performance boost to be had by using a single assertCodeExecution, instead of multiple. In case of an error, The diff will make it relatively easy to work out which line was the problem (and if it doesn't, you can modify the string output to put helpful markers to help navigate)
2. There's no test of the "default" behavior - where you *don't* specify a fill character
3. There's no test of the math when the input string is empty, or when the size is divisible by 2. Given that we're dealing with halving in the implementation, checking even and odd values is helpful.